hyperframe/frame.py

```diff
@@ -110,7 +113,15 @@ def parse_frame_header(header):
         stream_id = fields[4]
 
         if type not in FRAMES:
-            raise UnknownFrameError(type, length)
+            if strict:
+                raise UnknownFrameError(type, length)
+
+            frame = ExtensionFrame(
+                type=type,
+                flag_byte=flags,
+                stream_id=stream_id,
+            )
+            return (frame, length)
```

Comment:
So I am not sure that `assign_flag_mapping` is a good idea tbh. I think that realistically anyone who wants to interpret the body of a new frame has to handle basically everything except the parsing of the frame header. So I'd be happy to just have us save the flag_byte and call it a day.